### PR TITLE
[#2238] feat(server): Add the operations for the user

### PR DESCRIPTION
**Pull request by qqqttt123**
### What changes were proposed in this pull request?

Add the operations for the user.

### Why are the changes needed?

Fix: #2238

### Does this PR introduce _any_ user-facing change?
Yes,  I will add the open api and the document in the later pr.

### How was this patch tested?
Add a new UT.

### CODE REVIEW

File: server/src/main/java/com/datastrato/gravitino/server/web/rest/UserOperations.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  @ResponseMetered(name = "remove-user", absolute = true)
+  public Response removeUser(
+      @PathParam("metalake") String metalake, @PathParam("user") String user) {
+    try {
+      return Utils.doAs(
+          httpRequest,
+          () -> {
+            NameIdentifier ident = ofUser(metalake, user);
+            boolean removed =
+                TreeLockUtils.doWithTreeLock(
```
Comment (qqqttt123) [inline server/src/main/java/com/datastrato/gravitino/server/web/rest/UserOperations.java:107]: @yuqi1129 Could you help me review the use of lock?

Comment (jerryshao) [reply server/src/main/java/com/datastrato/gravitino/server/web/rest/UserOperations.java:107]: This part maybe changed after #2732 , let's wait for a while.

**Comment by jerryshao** [comment]
LGTM, @yuqi1129 would you please double the usage of tree lock here?

**Comment by yuqi1129** [comment]
> LGTM, @yuqi1129 would you please double the usage of tree lock here?

No problem was found here, so it's okay for me with the changes.

**Comment by jerryshao** [comment]
OK, I think we can change the whole lock mechanism in #2732 if necessary, we don't have to block this PR.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**